extensions/2.0/Vendor/EXT_3dtiles_feature_metadata/README.md

```diff
@@ -0,0 +1,1565 @@
+```json
+{
+  "meshes": [
+    {
+      "primitives": [
+        {
+          "attributes": {
+            "POSITION": 0,
+            "_FEATURE_ID_0": 1
+          },
+          "indices": 2,
+          "extensions": {
+            "EXT_3dtiles_feature_metadata": {
+              "featureLayers": [
+                {
+                  "featureTable": 0,
+                  "featureIds": {
+                    "attribute": "_FEATURE_ID_0"
+                  }
+                }
+              ]
+            }
+          }
+        }
+      ]
+    }
+  ],
+  "extensions": {
+    "EXT_3dtiles_feature_metadata": {
+      "featureTables": [
+        {
+          "featureCount": 2,
+          "featureProperties": {
+            "Name": {
+              "array": {
+                "type": "string",
+                "values": ["Building A", "Building B"]
+              }
+            },
+            "Year": {
+              "array": {
+                "type": "number",
+                "values": [1999, 2015]
+              }
+            },
+            "Coordinates": {
+              "accessor": 3
+            }
+          }
+        }
+      ]
+    }
+  },
+  "accessors": [
+    {
+      "name": "Positions",
+      "bufferView": 0,
+      "byteOffset": 0,
+      "componentType": 5126,
+      "count": 8,
+      "type": "VEC3"
+    },
+    {
+      "name": "Feature IDs",
+      "bufferView": 1,
+      "byteOffset": 0,
+      "componentType": 5126,
+      "count": 8,
+      "type": "SCALAR"
+    },
+    {
+      "name": "Indices",
+      "bufferView": 2,
+      "byteOffset": 0,
+      "componentType": 5123,
+      "count": 12,
+      "type": "SCALAR"
+    },
+    {
+      "name": "Coordinates (longitude, latitude)",
+      "bufferView": 3,
+      "byteOffset": 0,
+      "componentType": 5126,
+      "count": 2,
+      "type": "VEC2",
+      "max": [180, 90],
+      "min": [-180, -90]
+    }
+  ]
+}
+```
+
+## Concepts
+
+### Feature Layers
+
+Feature layers are per-primitive and define the mapping between vertices / texels and feature property data. They must always reference a feature table and may also encode feature property data directly. They contain the following properties:
+
+| Property            | Description                                                       | Caveats                                                                                                                                                                |
+|---------------------|-------------------------------------------------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `featureTable`      | Index of the feature table that this feature layer is using.      | Multiple feature layers in a single primitive cannot use the same feature table. Feature tables used by a single primitive cannot have any of the same property names. |
```

Comment:
> Multiple feature layers in a single primitive cannot use the same feature table.

Probably rare that this would be useful, but does it need to be forbidden?

> Feature tables used by a single primitive cannot have any of the same property names.

This seems like a really big limitation. Why is it necessary?

Reply:
> Probably rare that this would be useful, but does it need to be forbidden

I can't remember why anymore? But I could see a use case where a per-vertex ID layer and a per-texel ID layer want to use the same feature table. Some maybe we should lift this restriction.

> This seems like a really big limitation. Why is it necessary?

It was added for styling reasons to not overload variable names. Like `${Height}` could mean two different things if multiple feature tables have a Height property. Maybe this could be solved by prefixing variables with the feature table name like `${Buidings:Height}`, or something else. Styling is still a mostly unknown topic here. Personally I'd like to move closer to something like GLSL where conflicts like this can be worked around easier, and just gives more control in general. That really deserves its own issue.

Reply:
Despite the "3dtiles" in the name (maybe it shouldn't even be there?), I think of this extension as being pretty general. glTF currently lacks a way to attach metadata to a parts of a model, and that's useful in a whole lot of scenarios. This extension should work well for most of them. So we shouldn't let (current) limitations in the styling language impose limitations on metadata expression in this extension. Why not just define in the styling spec what happens when there's a conflict? Last one wins, or whatever.

Reply:
I removed both restrictions for now.

> Despite the "3dtiles" in the name (maybe it shouldn't even be there?)

I've seen some confusion about the naming recently. I added it to the TODO list. I'm curious to see what @pjcozzi thinks when he reviews.